stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/options/MessageOptionsView.kt

```diff
@@ -27,34 +28,82 @@ public class MessageOptionsView : FrameLayout {
         defStyleAttr
     )
 
-    internal fun configure(configuration: Configuration, isMessageTheirs: Boolean) {
+    internal fun configure(configuration: Configuration, isMessageTheirs: Boolean, syncStatus: SyncStatus) {
+        if (isMessageTheirs) {
+            configureTheirsMessage(configuration, syncStatus)
+        } else {
+            configureMineMessage(configuration, syncStatus)
+        }
+    }
+
+    private fun configureTheirsMessage(configuration: Configuration, syncStatus: SyncStatus) {
         val iconsTint = configuration.iconsTint
 
-        binding.replyTV.configureListItem(configuration.replyText, configuration.replyIcon, iconsTint)
-        binding.threadReplyTV.configureListItem(configuration.threadReplyText, configuration.threadReplyIcon, iconsTint)
+        // Todo: Uncomment once reply is done.
+        // binding.replyTV.configureListItem(configuration.replyText, configuration.replyIcon, iconsTint)
 
-        if (configuration.copyTextEnabled) {
-            binding.copyTV.isVisible = true
-            binding.copyTV.configureListItem(configuration.copyText, configuration.copyIcon, iconsTint)
-        } else {
-            binding.copyTV.isVisible = false
+        if (syncStatus == SyncStatus.COMPLETED) {
+            binding.threadReplyTV.configureListItem(
+                configuration.threadReplyText,
+                configuration.threadReplyIcon,
+                iconsTint
+            )
+        }
+
+        configureCopyMessage(iconsTint, configuration)
+
+        binding.flagTV.configureListItem(configuration.flagText, configuration.flagIcon, iconsTint)
+        binding.muteTV.configureListItem(configuration.muteText, configuration.muteIcon, iconsTint)
+        binding.blockTV.configureListItem(configuration.blockText, configuration.blockIcon, iconsTint)
+        binding.deleteTV.isVisible = false
+    }
+
+    private fun configureMineMessage(configuration: Configuration, syncStatus: SyncStatus) {
+        val iconsTint = configuration.iconsTint
+
+        // Todo: Uncomment once reply is done.
+        // binding.replyTV.configureListItem(configuration.replyText, configuration.replyIcon, iconsTint)
+
+        when (syncStatus) {
+            SyncStatus.FAILED_PERMANENTLY -> {
+                binding.retryTV.configureListItem(
+                    configuration.retryText,
+                    configuration.retryIcon,
+                    ContextCompat.getColor(context, R.color.stream_ui_blue)
+                )
+
+                binding.retryTV.isVisible = true
+                binding.threadReplyTV.isVisible = false
+            }
+            SyncStatus.COMPLETED -> {
+                binding.threadReplyTV.configureListItem(
+                    configuration.threadReplyText,
+                    configuration.threadReplyIcon,
+                    iconsTint
+                )
+            }
+            SyncStatus.SYNC_NEEDED, SyncStatus.IN_PROGRESS -> {
+            }
```

Comment:
What happens to the "thread reply" option in this case?

Reply:
It becomes not visible, but this is made in the XML in this situation. I'll do also move it to code.